### ISSUE TITLE
Quests: Design a way to incorporate quests and tasks into NPC struct and dialogue

### ISSUE THREAD

**Issue by elizabethli31** [uchicago-cs/chiventure#1468]
After meeting with NPC team, we were told how their dialogue works (triggers a conversation and actions based on key words). We had to come up with a way to have quests in certain dialogue options

**Comment by elizabethli31** [comment]
We decided to have npc_quest_list and npc_task_list where each node included an id and dialogue (node_t from npc's structs). If the player's stats meet the prereqs of the quests and tasks, then the NPC will go through with the dialogue that is stored in that particular node. We currently have functions that can help with this but would need the NPC team to actually write these functions. I wrote possible dialogue options and included all the function headers that would be involved with this task. https://github.com/uchicago-cs/chiventure/wiki/Quests-~-Incorporating-Quests-into-NPC
